rust/kernel/list/arc.rs

```diff
@@ -441,23 +443,8 @@ where
     }
 }
 
-// This is to allow coercion from `ListArc<T>` to `ListArc<U>` if `T` can be converted to the
-// dynamically-sized type (DST) `U`.
-impl<T, U, const ID: u64> core::ops::CoerceUnsized<ListArc<U, ID>> for ListArc<T, ID>
-where
-    T: ListArcSafe<ID> + Unsize<U> + ?Sized,
-    U: ListArcSafe<ID> + ?Sized,
-{
-}
-
-// This is to allow `ListArc<U>` to be dispatched on when `ListArc<T>` can be coerced into
-// `ListArc<U>`.
-impl<T, U, const ID: u64> core::ops::DispatchFromDyn<ListArc<U, ID>> for ListArc<T, ID>
-where
-    T: ListArcSafe<ID> + Unsize<U> + ?Sized,
-    U: ListArcSafe<ID> + ?Sized,
-{
-}
```

Comment:
We can't delete these due to support for older rustc releases. Please see [this doc](https://github.com/Rust-for-Linux/linux/blob/rust-next/Documentation/rust/general-information.rst#conditional-compilation) for info on conditional compilation.

Reply:
For reference, our current minimum version is 1.78.0, so you can use that to test.

Reply:
Okay, I will revert those chunks